src/nvim/ex_cmds.c

```diff
@@ -2946,6 +2951,20 @@ void do_sub(exarg_T *eap)
   int start_nsubs;
   int save_ma = 0;
 
+  //  list to save matched lines
+  //  Note: Could not use nolint with rule, line would be > 100 chars
+  klist_t(matchedline_T) *lmatch = kl_init(matchedline_T); // NOLINT
```

Comment:
Is there a reason why this line (if it needs to be in this function) cannot be moved in front of the big `for` loop? Then we would not need the `goto` replacement before the loop.